common/src/api/external/mod.rs

```diff
@@ -376,14 +376,33 @@ impl JsonSchema for NameOrId {
 
 // TODO: remove wrapper for semver::Version once this PR goes through
 // https://github.com/GREsau/schemars/pull/195
-#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Display)]
+#[derive(
+    Clone,
+    Debug,
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    Hash,
+    PartialOrd,
+    Ord,
+    Display,
+    FromStr,
+)]
 #[display("{0}")]
 pub struct SemverVersion(pub semver::Version);
 
 impl SemverVersion {
     pub fn new(major: u64, minor: u64, patch: u64) -> Self {
         Self(semver::Version::new(major, minor, patch))
     }
+
+    /// This is a permissive version of the official validation regex for
+    /// semver:
+    /// https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+    /// It accepts some inputs that are not valid semver. See
+    /// test::test_semver_validation for an example.
+    const VALIDATION_REGEX: &str = r"^\d+\.\d+\.\d+([\-\+].+)?$";
```

Comment:
I got that regex from [here](https://github.com/GREsau/schemars/pull/195/files#diff-3e6e1ed692353553259403498a33b36ca4920d868b70f8b1511ceb36be44110aR17) and didn't think about it as much as I should have. No reason we have to use this weaker regex instead of the semi-official ones.

Reply:
I switched over to the official validation regex.